controllers/kafkatopic_controller.go

```diff
@@ -32,10 +35,18 @@ import (
 	"github.com/banzaicloud/koperator/pkg/k8sutil"
 	"github.com/banzaicloud/koperator/pkg/kafkaclient"
 	"github.com/banzaicloud/koperator/pkg/util"
+	"github.com/banzaicloud/koperator/pkg/webhooks"
 )
 
 var topicFinalizer = "finalizer.kafkatopics.kafka.banzaicloud.io"
 
+func isTopicManagedByKoperator(topic metav1.Object) bool {
+	if manager, ok := topic.GetAnnotations()[webhooks.TopicManagedByAnnotationKey]; !ok || strings.ToLower(manager) == webhooks.TopicManagedByKoperatorAnnotationValue {
```

Comment:
**Recommended**: We **SHOULD NOT** return true for `!ok` case IMO, because in case there is no `managedBy` annotation on the object we should not assume Koperator created it. To my understanding what you wrote in the description (that the user needs to set the annotation if they want it to be managed) is a good way to handle this. Or am I misunderstanding something here?

```suggestion
	if manager, ok := topic.GetAnnotations()[webhooks.TopicManagedByAnnotationKey]; ok && strings.ToLower(manager) == webhooks.TopicManagedByKoperatorAnnotationValue {
```

Reply:
This is a little bit different from what we need in the kafkatopic_validation. This means that isTopicManagedByKoperator
thus when there is no annotation then it is created by koperator (because when there is existing topic and we just create kafkatopic cr we dont have to put annotation), and when it has annotation managedBy:koperator it is also managed by koperator so we need to apply changes based on the kafkatopic CR settings.